src/Microsoft.Health.Dicom.Blob/Features/ExternalStore/ExternalBlobClient.cs

```diff
@@ -0,0 +1,73 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+
+using Azure;
+using Azure.Core;
+using Azure.Storage.Blobs;
+using EnsureThat;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Blob.Configs;
+using Microsoft.Health.Dicom.Blob.Utilities;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Blob.Features.Storage;
+using Microsoft.Health.Dicom.Core.Exceptions;
+
+namespace Microsoft.Health.Dicom.Blob.Features.ExternalStore;
+
+/// Represents the blob container created by the user and initialized JIT
+internal class ExternalBlobClient : IBlobClient
+{
+    private readonly object _lockObj = new object();
+    private readonly BlobServiceClientOptions _blobClientOptions;
+    private readonly ExternalBlobDataStoreConfiguration _externalStoreOptions;
+    private readonly IExternalOperationCredentialProvider _credentialProvider;
+    private BlobContainerClient _blobContainerClient;
+
+    public ExternalBlobClient(
+        IExternalOperationCredentialProvider credentialProvider,
+        IOptions<ExternalBlobDataStoreConfiguration> externalStoreOptions,
+        IOptions<BlobServiceClientOptions> blobClientOptions)
+    {
+        _credentialProvider = EnsureArg.IsNotNull(credentialProvider, nameof(credentialProvider));
+        _blobClientOptions = EnsureArg.IsNotNull(blobClientOptions?.Value, nameof(blobClientOptions));
+        _externalStoreOptions = EnsureArg.IsNotNull(externalStoreOptions?.Value, nameof(externalStoreOptions));
+    }
+
+    public bool IsExternal => true;
+
+    public BlobContainerClient BlobContainerClient
```

Comment:
Why lazily create this?

Reply:
To allow container to initialize even when the BYOS is down

Reply:
I don't think any remote operation happens within the ctor for the client, so I think you're safe to create the client (unless the URL is not in the proper format).

Reply:
We will try to get token which access their key valut that can fail

Reply:
We're only fetching the [`TokenCredential`](https://learn.microsoft.com/en-us/dotnet/api/azure.core.tokencredential?view=azure-dotnet) I think which confusingly does not actually have the token. Instead, it just has a method to fetch the token like `GetTokenAsync`

Reply:
I have also see Verify checks on the BlobContainerClient https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs.

This maybe pessimestic, but I would prefer this code to be lazy and catch and throw rather than failing the app intialization. I don't think there is any con with being lazy other than the lock logic.

Reply:
Linked AB#102834 for related design discussion.